service/internal/credentials/vault.go

```diff
@@ -432,6 +441,33 @@ func (v VaultProvider) isAdmin() bool {
 	return v.roleID == authorizationKeyAdmin
 }
 
+func (v VaultProvider) ListProjectTokens(projectName string) ([]types.ProjectToken, error) {
+	projectTokens := make([]types.ProjectToken, 0)
+	if !v.isAdmin() {
+		return nil, errors.New("admin credentials must be used to list project tokens")
+	}
+
+	path := fmt.Sprintf("%s/secret-id", genProjectAppRole(projectName))
+	tokens, err := v.vaultLogicalSvc.List(path)
+	if err != nil {
+		return nil, fmt.Errorf("vault list error: %w", err)
+	}
+
+	if _, ok := tokens.Data["keys"]; !ok {
+		return projectTokens, nil
+	}
+
+	for _, key := range tokens.Data["keys"].([]interface{}) {
+		tok := types.ProjectToken{
+			ID: key.(string),
```

Comment:
If we make a call to `GetProjectTokens()` here, we can populate object the create and expire times. This will clean up the handler code a bit, and caller won't have to make another call (`getToken`) to get those times.

Reply:
I support this, let's populate the objects from Vault data